src/main/java/com/itspace/job/service/UserServiceImpl.java

```diff
@@ -19,28 +21,48 @@
     private final UserRoleRepository userRoleRepository;
     private final PasswordEncoder passwordEncoder;
     private final ModelMapper mapper;
+    private final JwtUtil jwtUtil;
 
-    public UserServiceImpl(UserRepository userRepository, UserRoleRepository userRoleRepository, PasswordEncoder passwordEncoder, ModelMapper mapper) {
+    public UserServiceImpl(UserRepository userRepository, UserRoleRepository userRoleRepository, PasswordEncoder passwordEncoder, ModelMapper mapper, JwtUtil jwtUtil) {
         this.userRepository = userRepository;
         this.userRoleRepository = userRoleRepository;
         this.passwordEncoder = passwordEncoder;
         this.mapper = mapper;
+        this.jwtUtil = jwtUtil;
     }
 
-
     @Override
     public User register(RegisterUserDto registerUserDto) {
         User byEmail = userRepository.findByEmail(registerUserDto.getEmail());
         User user = new User();
         if (byEmail == null) {
+            mapper.map(registerUserDto, user);
             user.setPassword(passwordEncoder.encode(registerUserDto.getPassword()));
             UserRole userRoleByName = userRoleRepository.findByName("user");
             List<UserRole> all = new ArrayList<>();
             all.add(userRoleByName);
             user.setUserRoles(all);
-            mapper.map(registerUserDto,user);
             return userRepository.save(user);
         }
         return null;
     }
+
+    @Override
+    public LoginResponseDto login(LoginUser loginUser) {
+        User byEmail = userRepository.findByEmail(loginUser.getEmail());
+        if (byEmail != null && passwordEncoder.matches(loginUser.getPassword(), byEmail.getPassword())){
+            String token = jwtUtil.generateToken(loginUser.getEmail(), new DefaultClaims());
+            UserDto userDto = mapper.map(byEmail,UserDto.class);
+            userDto.setUserRole(new ArrayList<>());
+            for (UserRole userRole : byEmail.getUserRoles()) {
```

Comment:
please create UserRoleDto and set this dto in the UserDto class instead of UserRole entity.
and after that you can delete this code where you set userRoles manually.
ModelMapper will do it.
This line (mapper.map(byEmail,UserDto.class);) is enough, which you have already written.
thank you.